File: seed/csharp-model/trace/src/SeedTraceClient/V2/V3/Problem/FileInfoV2.cs
using System.Text.Json.Serialization

namespace SeedTraceClient.V2.V3

public class FileInfoV2
{
    [JsonPropertyName("filename")]
    public string Filename { get; init; }

    [JsonPropertyName("directory")]
    public string Directory { get; init; }

    [JsonPropertyName("contents")]
    public string Contents { get; init; }

    [JsonPropertyName("editable")]
    public bool Editable { get; init; }
}

File: seed/rust-sdk/trace/src/types/file_info_v_2.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
pub struct FileInfoV2 {
    pub filename: String,
    pub directory: String,
    pub contents: String,
    pub editable: bool,
}